File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"  // equal to "from example_interfaces.msg import String"

using namespace rclcpp;
using namespace std;

class RobotNewsStationNode : public Node 
{
public:
    RobotNewsStationNode() : Node("robot_news_station") 
    {
        this->declare_parameter("robot_name","C3D3");
        robot_names_ = this->get_parameter("robot_name").as_string();
        publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news",10);
        timer_ = this->create_wall_timer(chrono::milliseconds(500),
                                            bind(&RobotNewsStationNode::publishNews,this));
        RCLCPP_INFO(this->get_logger(),"Robot News Station has been started.");
    }

private:
    void publishNews()
    {
        auto msg = example_interfaces::msg::String();       //use String type in example_interfaces/msg
        msg.data = string("Hi, this is ") + robot_names_ + string(" from the Robot News Station");
        publisher_->publish(msg);
    }
    string robot_names_;
    Publisher<example_interfaces::msg::String>::SharedPtr publisher_;       //publisher_ : SharedPtr
    TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>(); 
    spin(node);
    shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String

class RobotNewsStationNode(Node): 
    def __init__(self):
        super().__init__("robot_news_station")
        self.robot_name_ = "C3PO"
        self.publishers_ = self.create_publisher(String,"robot_news",10)        #(msg_type,topic_name,buffer)
        self.timer_ = self.create_timer(0.5,self.publish_news)
        self.get_logger().info("Robot news has been started")

    def publish_news(self):
        msg = String()      #String object
        msg.data = "Hi, this is " + self.robot_name_ + " from the robot news station"
        self.publishers_.publish(msg)
        

def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStationNode() 
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()